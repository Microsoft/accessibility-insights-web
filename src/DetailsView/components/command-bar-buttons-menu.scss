// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

.command-bar-buttons-menu-button {
    height: inherit;
}

.command-bar-buttons-submenu {
    button {
        height: 36px;
    }
}
