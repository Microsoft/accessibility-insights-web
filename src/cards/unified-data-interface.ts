// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// this is similar to `TestEngine` interface from axe-core
interface ScanEngine {
    name: string;
    version: string;
}

interface OSData {
    name: string;
    osVersion?: string;
    userAgent?: string;
}

interface ViewPortData {
    platformHeight?: number;
    platformWidth?: number;
}

interface PlatformData {
    osInfo: OSData;
    viewPortInfo: ViewPortData;
}

interface ToolData {
    scanEngine: ScanEngine;
}

interface ScanMetaData {}

interface ScanResults {
    results: UnifiedResults;
    platformInfo: PlatformData;
    toolInfo: ToolData;
    scanMetaData: ScanMetaData;
}

interface UnifiedResults {
    passes: UnifiedResultInstance[];
    violations: UnifiedResultInstance[];
    incomplete: UnifiedResultInstance[];
}

interface UnifiedRuleResult {
    id: string;
    ruleDescription: string;
}

interface InstancePropertyBag<T> {
    [property: string]: T;
}

type StoredInstancePropertyBag = InstancePropertyBag<any>;

interface UnifiedResultInstance {
    id: string;
    evaluation: StoredInstancePropertyBag;
    identifiers: StoredInstancePropertyBag;
    descriptors: StoredInstancePropertyBag;
    resolution: StoredInstancePropertyBag;
}
