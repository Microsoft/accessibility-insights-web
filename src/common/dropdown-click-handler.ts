// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { DropdownActionMessageCreator } from './message-creators/dropdown-action-message-creator';
import { TelemetryEventSource } from './types/telemetry-data';

export class DropdownClickHandler {
    private dropdownActionMessageCreator: DropdownActionMessageCreator;
    private source: TelemetryEventSource;

    constructor(
        dropdownActionMessageCreator: DropdownActionMessageCreator,
        source: TelemetryEventSource,
    ) {
        this.source = source;
        this.dropdownActionMessageCreator = dropdownActionMessageCreator;
    }

    public openPreviewFeaturesPanelHandler = (event: React.MouseEvent<HTMLElement>): void => {
        this.dropdownActionMessageCreator.openPreviewFeaturesPanel(event, this.source);
    };

    public openScopingPanelHandler = (event: React.MouseEvent<HTMLElement>): void => {
        this.dropdownActionMessageCreator.openScopingPanel(event, this.source);
    };

    public openSettingsPanelHandler = (event: React.MouseEvent<HTMLElement>): void => {
        this.dropdownActionMessageCreator.openSettingsPanel(event, this.source);
    };

    public openDebugTools = (): void => {
        this.dropdownActionMessageCreator.openDebugTools();
    };
}
