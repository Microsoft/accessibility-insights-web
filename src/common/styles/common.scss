// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import 'high-contrast-override.scss';
@import 'colors.scss';
@import 'fonts.scss';

$focus-ring-color: $neutral-100;

@mixin set-focus-ring-styles($property: 'border') {
    #{$property}: 1px solid $focus-ring-color;
    @include ms-high-contrast-override {
        #{$property}: 1px dashed windowtext !important;
    }
}

.ms-Fabric.is-focusVisible .ms-Nav-group .ms-nav-linkButton:focus::after {
    @include set-focus-ring-styles('border');
}

button::after {
    content: '';
    position: absolute;
}

.ms-Button.ms-Button--primary:not(.is-disabled) {
    background: $communication-primary;

    &:hover {
        background: $communication-shade-10;
    }
}

.ms-fontColor-neutralPrimary,
.ms-fontColor-neutralPrimary--hover:hover {
    color: $neutral-100 !important;
}

.ms-Fabric {
    color: $neutral-100;
}

.generic-toggle-component {
    margin-bottom: 4vh;
    .toggle-container {
        margin-bottom: 1vh;
        .toggle-name {
            display: inline-block;
            font-size: 17px;
            font-weight: bold;
        }

        .toggle {
            float: right;
        }
    }
}

.high-contrast-theme {
    .insights-link {
        &:hover {
            text-decoration: underline;
        }
    }
    .is-selected {
        .status-icon {
            color: $high-contrast-icon-color !important;
        }
    }
}

.insights-link {
    // marking important to override office fabric style
    color: $link !important;
    text-decoration: none;

    &:hover {
        color: $link-hover !important;
        text-decoration: underline;
    }
}

.ms-Spinner {
    .ms-Spinner-circle {
        border-top-color: $spinner-text;
    }

    .ms-Spinner-label {
        color: $spinner-text;
    }
}

$normalTitleFontWeight: 300 !default;

$pivotItemLeftBorderWidth: 3px !default;
$pivotItemBorderStyle: solid !default;

$detailsViewHeaderBarHeight: 40px !default;
$detailsViewNavPivotsHeight: 41px !default;
$detailsViewCommandBarHeight: 40px !default;
$detailsViewCommandBarBorderHeight: 1px !default;
$detailsViewLeftNavWidth: 200px !default;
$detailsViewTotalHeaderHeight: (#{$detailsViewCommandBarBorderHeight} + #{$detailsViewCommandBarHeight} + #{$detailsViewHeaderBarHeight});

$fastPassRightPanelMarginRight: 24px !default;
$fastPassRightPanelMarginLeft: 24px !default;
$fastPassRightPanelMarginTop: 24px !default;

@mixin ellipsedText {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
