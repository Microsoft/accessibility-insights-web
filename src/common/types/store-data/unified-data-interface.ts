// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GuidanceLink } from '../../../scanner/rule-to-links-mappings';

// this is similar to `TestEngine` interface from axe-core
export interface ScanEngineProperties {
    name: string;
    version: string;
}

export interface ApplicationProperties {
    name: string;
    version: string;
}

export interface OSProperties {
    name: string;
    version: string;
    userAgent?: string;
}

export interface ViewPortProperties {
    height?: number;
    width?: number;
}

export interface PlatformData {
    osInfo: OSProperties;
    viewPortInfo: ViewPortProperties;
}

export interface TargetAppData {
    name: string;
    url?: string;
}

export interface ToolData {
    scanEngineProperties: ScanEngineProperties;
    applicationProperties: ApplicationProperties;
}

export interface UnifiedRule {
    id: string;
    description: string;
    url: string;
    guidance: GuidanceLink[];
}

export interface UnifiedScanResultStoreData {
    results: UnifiedResult[];
    rules: UnifiedRule[];
    platformInfo?: PlatformData;
    toolInfo?: ToolData;
    targetAppInfo?: TargetAppData;
}

export interface InstancePropertyBag {
    [property: string]: any;
}

export type StoredInstancePropertyBag = InstancePropertyBag;

export type UnifiedIdentifiers = {
    identifier: string;
    conciseName: string;
} & InstancePropertyBag;

export type UnifiedDescriptors = {
    snippet?: string;
} & InstancePropertyBag;

export type UnifiedResolution = {
    howToFixSummary: string;
} & InstancePropertyBag;

export interface UnifiedResult {
    uid: string;
    status: InstanceResultStatus;
    ruleId: string;
    identifiers: UnifiedIdentifiers;
    descriptors: UnifiedDescriptors;
    resolution: UnifiedResolution;
}

export type InstanceResultStatus = 'pass' | 'fail' | 'unknown';

export type UnifiedRuleResultStatus = InstanceResultStatus | 'inapplicable';

export interface UnifiedRuleResult {
    id: string;
    nodes: UnifiedResult[];
    description: string;
    url: string;
    guidance: GuidanceLink[];
}

export type UnifiedStatusResults = {
    [key in UnifiedRuleResultStatus]: UnifiedRuleResult[];
};

export const AllRuleResultStatuses: UnifiedRuleResultStatus[] = ['pass', 'fail', 'unknown', 'inapplicable'];
