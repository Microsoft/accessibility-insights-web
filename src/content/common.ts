// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { link } from './link';
import { ContentCreator } from '../views/content/content-page';
import { GuidanceTitle as gt } from '../views/content/guidance-title';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
