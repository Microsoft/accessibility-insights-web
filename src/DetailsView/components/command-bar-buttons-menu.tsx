// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { CommandBarProps } from 'DetailsView/components/details-view-command-bar';
import { CommandBarButton, IOverflowSetItemProps, OverflowSet } from 'office-ui-fabric-react';
import * as React from 'react';
import * as styles from './command-bar-buttons-menu.scss';

export type CommandBarButtonsMenuProps = CommandBarProps;

export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
    'CommandBarButtonsMenu',
    props => {
        const onRenderItem = (item: IOverflowSetItemProps): JSX.Element => {
            return item.onRender(props);
        };

        const onRenderOverflowButton = (overflow: IOverflowSetItemProps[]): JSX.Element => {
            return (
                <CommandBarButton
                    ariaLabel="More items"
                    role="menuitem"
                    menuIconProps={{
                        iconName: 'More',
                        className: styles.commandBarButtonsMenuButton,
                    }}
                    menuProps={{ items: overflow, className: styles.commandBarButtonsSubmenu }}
                />
            );
        };

        const overflowItems = [
            {
                key: 'export report',
                onRender: () => props.switcherNavConfiguration.ReportExportComponentFactory(props),
            },
            {
                key: 'start over',
                onRender: () =>
                    props.switcherNavConfiguration.StartOverComponentFactory({
                        ...props,
                        dropdownDirection: 'left',
                    }),
            },
        ];

        return (
            <OverflowSet
                onRenderItem={onRenderItem}
                overflowItems={overflowItems}
                onRenderOverflowButton={onRenderOverflowButton}
                className={styles.commandBarButtonsMenu}
            />
        );
    },
);
