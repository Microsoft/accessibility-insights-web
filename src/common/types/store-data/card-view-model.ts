// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GuidanceLink } from '../../../scanner/rule-to-links-mappings';
import { InstanceResultStatus, UnifiedResult } from './unified-data-interface';
export type CardRuleResultStatus = InstanceResultStatus | 'inapplicable';
export interface CardRuleResult {
    id: string;
    nodes: CardResult[];
    description: string;
    url: string;
    guidance: GuidanceLink[];
    isExpanded: boolean;
}
export type CardRuleResultsByStatus = {
    [key in CardRuleResultStatus]: CardRuleResult[];
};
export interface CardResult extends UnifiedResult {
    isSelected: boolean;
}
export const AllRuleResultStatuses: CardRuleResultStatus[] = ['pass', 'fail', 'unknown', 'inapplicable'];
