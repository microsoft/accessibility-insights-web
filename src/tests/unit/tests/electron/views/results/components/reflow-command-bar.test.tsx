// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DropdownClickHandler } from 'common/dropdown-click-handler';
import { CardsViewModel } from 'common/types/store-data/card-view-model';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { ScanMetadata, ToolData } from 'common/types/store-data/unified-data-interface';
import { CommandBarButtonsMenu } from 'DetailsView/components/command-bar-buttons-menu';
import { NarrowModeStatus } from 'DetailsView/components/narrow-mode-detector';
import { ScanStoreData } from 'electron/flux/types/scan-store-data';
import { ContentPageInfo } from 'electron/types/content-page-info';
import {
    commandButtonRefreshId,
    ReflowCommandBar,
    ReflowCommandBarDeps,
    ReflowCommandBarProps,
} from 'electron/views/results/components/reflow-command-bar';
import { mount, shallow } from 'enzyme';
import { isMatch } from 'lodash';
import { IButton } from 'office-ui-fabric-react';
import * as React from 'react';
import { ReportGenerator } from 'reports/report-generator';
import { getAutomationIdSelector } from 'tests/common/get-automation-id-selector';
import { EventStubFactory } from 'tests/unit/common/event-stub-factory';
import { IMock, It, Mock, Times } from 'typemoq';

describe('ReflowCommandBar', () => {
    let featureFlagStoreDataStub: FeatureFlagStoreData;
    let cardsViewDataStub: CardsViewModel;
    let reportGeneratorMock: IMock<ReportGenerator>;
    let scanMetadataStub: ScanMetadata;
    let scanDateStub: Date;
    let narrowModeStatusStub: NarrowModeStatus;
    let props: ReflowCommandBarProps;

    beforeEach(() => {
        featureFlagStoreDataStub = {
            somefeatureflag: true,
        };
        cardsViewDataStub = {} as CardsViewModel;
        scanMetadataStub = {
            timespan: {
                scanComplete: scanDateStub,
            },
            toolData: {} as ToolData,
            targetAppInfo: {
                name: 'scan target name',
            },
        };
        narrowModeStatusStub = {
            isHeaderAndNavCollapsed: false,
            isCommandBarCollapsed: false,
            isVirtualKeyboardCollapsed: false,
        };
        scanDateStub = new Date(0);
        reportGeneratorMock = Mock.ofType(ReportGenerator);

        props = {
            deps: {
                scanActionCreator: null,
                reportGenerator: reportGeneratorMock.object,
            } as ReflowCommandBarDeps,
            scanStoreData: {} as ScanStoreData,
            cardsViewData: cardsViewDataStub,
            featureFlagStoreData: featureFlagStoreDataStub,
            scanMetadata: scanMetadataStub,
            narrowModeStatus: narrowModeStatusStub,
            isSideNavOpen: true,
            setSideNavOpen: () => null,
            currentContentPageInfo: {
                allowsExportReport: true,
                startOverButtonSettings: _ => {
                    return {
                        onClick: e => {},
                        disabled: false,
                    };
                },
            } as ContentPageInfo,
        };
    });

    describe('renders', () => {
        it('with start over button disabled', () => {
            props.currentContentPageInfo.startOverButtonSettings = _ => {
                return {
                    onClick: () => {},
                    disabled: true,
                };
            };

            const rendered = shallow(<ReflowCommandBar {...props} />);

            expect(rendered.getElement()).toMatchSnapshot();
        });

        it('does not create report export when scan metadata is null', () => {
            props.scanMetadata = null;
            const rendered = shallow(<ReflowCommandBar {...props} />);

            expect(rendered.getElement()).toMatchSnapshot();
        });

        it('does not create report export when allowsExportReport is false', () => {
            props.currentContentPageInfo.allowsExportReport = false;
            const rendered = shallow(<ReflowCommandBar {...props} />);

            expect(rendered.getElement()).toMatchSnapshot();
        });
    });

    describe('user interaction', () => {
        const eventStub = new EventStubFactory().createMouseClickEvent();

        it('handles start over click', () => {
            let clickWasCalled: boolean = false;

            props.currentContentPageInfo.startOverButtonSettings = _ => {
                return {
                    onClick: e => {
                        clickWasCalled = true;

                        // The event passed through is decorated with other properties so we only check subset.
                        expect(isMatch(e, eventStub)).toBeTruthy();
                    },
                    disabled: false,
                };
            };

            const rendered = mount(<ReflowCommandBar {...props} />);

            const buttonSelector = `button${getAutomationIdSelector(commandButtonRefreshId)}`;
            const button = rendered.find(buttonSelector);

            button.simulate('click', eventStub);

            expect(clickWasCalled).toBeTruthy();
        });

        it('handles settings click', () => {
            const dropdownClickHandlerMock = Mock.ofType<DropdownClickHandler>();
            props.deps.dropdownClickHandler = dropdownClickHandlerMock.object;

            const rendered = mount(<ReflowCommandBar {...props} />);
            const button = rendered.find('button[aria-label="settings"]');

            button.simulate('click', eventStub);

            dropdownClickHandlerMock.verify(
                handler => handler.openSettingsPanelHandler(It.isAny()),
                Times.once(),
            );
        });
    });

    describe('reflow behavior', () => {
        test('isCommandBarCollapsed is true', () => {
            props.narrowModeStatus.isCommandBarCollapsed = true;

            const rendered = shallow(<ReflowCommandBar {...props} />);
            const commandBar = rendered.find(CommandBarButtonsMenu);

            expect(rendered.getElement()).toMatchSnapshot();
            expect(commandBar.prop('renderExportReportButton')()).toMatchSnapshot('export report');
        });

        test('isHeaderAndNavCollapsed is true', () => {
            props.narrowModeStatus.isHeaderAndNavCollapsed = true;

            const rendered = shallow(<ReflowCommandBar {...props} />);

            expect(rendered.getElement()).toMatchSnapshot();
        });

        test('dropdown menu is dismissed and button focused when dialog is dismissed', () => {
            props.narrowModeStatus.isCommandBarCollapsed = true;
            const rendered = shallow(<ReflowCommandBar {...props} />);
            const buttonMock = Mock.ofType<IButton>();
            const commandBar = rendered.find(CommandBarButtonsMenu);
            const buttonRefCallback = commandBar.prop('buttonRef') as any;
            const onDialogDismissCallback = commandBar.prop('renderExportReportButton')().props[
                'onDialogDismiss'
            ];

            buttonMock.setup(bm => bm.dismissMenu()).verifiable();
            buttonMock.setup(bm => bm.focus()).verifiable();

            buttonRefCallback(buttonMock.object);
            onDialogDismissCallback();

            buttonMock.verifyAll();
        });
    });
});
