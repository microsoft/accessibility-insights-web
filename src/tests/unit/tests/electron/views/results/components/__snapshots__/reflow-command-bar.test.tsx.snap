// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReflowCommandBar reflow behavior isCommandBarCollapsed is true 1`] = `
<section
  aria-label="command bar"
  className="commandBar"
>
  <div
    className="farItems reflow"
  >
    <CommandBarButtonsMenu
      buttonRef={[Function]}
      getStartOverMenuItem={[Function]}
      renderExportReportButton={[Function]}
    />
    <InsightsCommandButton
      ariaLabel="settings"
      className="settingsGearButton"
      data-automation-id="command-button-settings"
      iconProps={
        Object {
          "className": "settingsGearButtonIcon",
          "iconName": "Gear",
        }
      }
      onClick={[Function]}
    />
  </div>
</section>
`;

exports[`ReflowCommandBar reflow behavior isCommandBarCollapsed is true: export report 1`] = `
<ReportExportComponent
  deps={
    Object {
      "reportExportServiceProvider": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "services": undefined,
      },
      "reportGenerator": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "assessmentJsonExportGenerator": undefined,
        "assessmentReportHtmlGenerator": undefined,
        "reportHtmlGenerator": undefined,
        "reportNameGenerator": undefined,
      },
      "scanActionCreator": null,
    }
  }
  featureFlagStoreData={
    Object {
      "somefeatureflag": true,
    }
  }
  getExportDescription={[Function]}
  htmlGenerator={[Function]}
  jsonGenerator={[Function]}
  onDialogDismiss={[Function]}
  pageTitle="scan target name"
  reportExportFormat="AutomatedChecks"
  reportExportServices={
    Array [
      Object {
        "generateMenuItem": null,
        "key": "html",
      },
      Object {
        "generateMenuItem": null,
        "key": "codepen",
      },
    ]
  }
  scanDate={1970-01-01T00:00:00.000Z}
  updatePersistedDescription={[Function]}
/>
`;

exports[`ReflowCommandBar reflow behavior isHeaderAndNavCollapsed is true 1`] = `
<section
  aria-label="command bar"
  className="commandBar"
>
  <FastPassLeftNavHamburgerButton
    className="navMenu"
    isSideNavOpen={true}
    setSideNavOpen={[Function]}
  />
  <div
    className="farItems reflow"
  >
    <React.Fragment>
      <FlaggedComponent
        enableJSXElement={
          <ReportExportComponent
            deps={
              Object {
                "reportExportServiceProvider": proxy {
                  "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
                  "services": undefined,
                },
                "reportGenerator": proxy {
                  "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
                  "assessmentJsonExportGenerator": undefined,
                  "assessmentReportHtmlGenerator": undefined,
                  "reportHtmlGenerator": undefined,
                  "reportNameGenerator": undefined,
                },
                "scanActionCreator": null,
              }
            }
            featureFlagStoreData={
              Object {
                "somefeatureflag": true,
              }
            }
            getExportDescription={[Function]}
            htmlGenerator={[Function]}
            jsonGenerator={[Function]}
            onDialogDismiss={[Function]}
            pageTitle="scan target name"
            reportExportFormat="AutomatedChecks"
            reportExportServices={
              Array [
                Object {
                  "generateMenuItem": null,
                  "key": "html",
                },
                Object {
                  "generateMenuItem": null,
                  "key": "codepen",
                },
              ]
            }
            scanDate={1970-01-01T00:00:00.000Z}
            updatePersistedDescription={[Function]}
          />
        }
        featureFlag="exportReport"
        featureFlagStoreData={
          Object {
            "somefeatureflag": true,
          }
        }
      />
      <InsightsCommandButton
        data-automation-id="command-button-refresh"
        disabled={false}
        iconProps={
          Object {
            "iconName": "Refresh",
          }
        }
        onClick={[Function]}
        text="Start over"
      />
    </React.Fragment>
    <InsightsCommandButton
      ariaLabel="settings"
      className="settingsGearButton"
      data-automation-id="command-button-settings"
      iconProps={
        Object {
          "className": "settingsGearButtonIcon",
          "iconName": "Gear",
        }
      }
      onClick={[Function]}
    />
  </div>
</section>
`;

exports[`ReflowCommandBar renders does not create report export when allowsExportReport is false 1`] = `
<section
  aria-label="command bar"
  className="commandBar"
>
  <div
    className="farItems reflow"
  >
    <React.Fragment>
      <FlaggedComponent
        enableJSXElement={null}
        featureFlag="exportReport"
        featureFlagStoreData={
          Object {
            "somefeatureflag": true,
          }
        }
      />
      <InsightsCommandButton
        data-automation-id="command-button-refresh"
        disabled={false}
        iconProps={
          Object {
            "iconName": "Refresh",
          }
        }
        onClick={[Function]}
        text="Start over"
      />
    </React.Fragment>
    <InsightsCommandButton
      ariaLabel="settings"
      className="settingsGearButton"
      data-automation-id="command-button-settings"
      iconProps={
        Object {
          "className": "settingsGearButtonIcon",
          "iconName": "Gear",
        }
      }
      onClick={[Function]}
    />
  </div>
</section>
`;

exports[`ReflowCommandBar renders does not create report export when scan metadata is null 1`] = `
<section
  aria-label="command bar"
  className="commandBar"
>
  <div
    className="farItems reflow"
  >
    <React.Fragment>
      <FlaggedComponent
        enableJSXElement={null}
        featureFlag="exportReport"
        featureFlagStoreData={
          Object {
            "somefeatureflag": true,
          }
        }
      />
      <InsightsCommandButton
        data-automation-id="command-button-refresh"
        disabled={false}
        iconProps={
          Object {
            "iconName": "Refresh",
          }
        }
        onClick={[Function]}
        text="Start over"
      />
    </React.Fragment>
    <InsightsCommandButton
      ariaLabel="settings"
      className="settingsGearButton"
      data-automation-id="command-button-settings"
      iconProps={
        Object {
          "className": "settingsGearButtonIcon",
          "iconName": "Gear",
        }
      }
      onClick={[Function]}
    />
  </div>
</section>
`;

exports[`ReflowCommandBar renders with start over button disabled 1`] = `
<section
  aria-label="command bar"
  className="commandBar"
>
  <div
    className="farItems reflow"
  >
    <React.Fragment>
      <FlaggedComponent
        enableJSXElement={
          <ReportExportComponent
            deps={
              Object {
                "reportExportServiceProvider": proxy {
                  "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
                  "services": undefined,
                },
                "reportGenerator": proxy {
                  "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
                  "assessmentJsonExportGenerator": undefined,
                  "assessmentReportHtmlGenerator": undefined,
                  "reportHtmlGenerator": undefined,
                  "reportNameGenerator": undefined,
                },
                "scanActionCreator": null,
              }
            }
            featureFlagStoreData={
              Object {
                "somefeatureflag": true,
              }
            }
            getExportDescription={[Function]}
            htmlGenerator={[Function]}
            jsonGenerator={[Function]}
            onDialogDismiss={[Function]}
            pageTitle="scan target name"
            reportExportFormat="AutomatedChecks"
            reportExportServices={
              Array [
                Object {
                  "generateMenuItem": null,
                  "key": "html",
                },
                Object {
                  "generateMenuItem": null,
                  "key": "codepen",
                },
              ]
            }
            updatePersistedDescription={[Function]}
          />
        }
        featureFlag="exportReport"
        featureFlagStoreData={
          Object {
            "somefeatureflag": true,
          }
        }
      />
      <InsightsCommandButton
        data-automation-id="command-button-refresh"
        disabled={true}
        iconProps={
          Object {
            "iconName": "Refresh",
          }
        }
        onClick={[Function]}
        text="Start over"
      />
    </React.Fragment>
    <InsightsCommandButton
      ariaLabel="settings"
      className="settingsGearButton"
      data-automation-id="command-button-settings"
      iconProps={
        Object {
          "className": "settingsGearButtonIcon",
          "iconName": "Gear",
        }
      }
      onClick={[Function]}
    />
  </div>
</section>
`;
